src/pubsub/ua_pubsub_manager.c

```diff
@@ -48,6 +48,23 @@ UA_Server_addPubSubConnection(UA_Server *server,
         return retval;
     }
 
+    if(UA_Variant_isEmpty(&tmpConnectionConfig->publisherId)){
+        UA_Byte publisherId = 0;
+        UA_Variant_setScalarCopy(&tmpConnectionConfig->publisherId, &publisherId, &UA_TYPES[UA_TYPES_BYTE]);
+    }
+
+    if(!(tmpConnectionConfig->publisherId.type == &UA_TYPES[UA_TYPES_BYTE] ||
+    tmpConnectionConfig->publisherId.type == &UA_TYPES[UA_TYPES_UINT16] ||
+    tmpConnectionConfig->publisherId.type == &UA_TYPES[UA_TYPES_UINT32] ||
+    tmpConnectionConfig->publisherId.type == &UA_TYPES[UA_TYPES_UINT64] ||
+    tmpConnectionConfig->publisherId.type == &UA_TYPES[UA_TYPES_STRING])){
+        UA_PubSubConnectionConfig_clear(tmpConnectionConfig);
+        UA_free(tmpConnectionConfig);
+        UA_LOG_ERROR(&server->config.logger, UA_LOGCATEGORY_SERVER,
+                     "PubSub Connection creation failed. Invalid PublisherIdType.");
+        return UA_STATUSCODE_BADTYPEMISMATCH;
+    }
```

Comment:
It is also better to have a similar check inside addDataSetReader API.

Reply:
Yes thank you that makes sense, i allowed empty Variants, which represent the NULL value, but this needs the implementation of processing Networkmessages without publisherId on either publisher or subscriber side, for it not to be caught in BADINTERNALERROR.